beacon_node/execution_layer/src/engines.rs

```diff
@@ -179,164 +167,112 @@ impl Engine {
     pub async fn is_synced(&self) -> bool {
         *self.state.read().await == EngineState::Synced
     }
+
     /// Run the `EngineApi::upcheck` function if the node's last known state is not synced. This
     /// might be used to recover the node if offline.
-    pub async fn upcheck_not_synced(&self, logging: Logging) {
-        let mut state_lock = self.state.write().await;
-        if *state_lock != EngineState::Synced {
-            match self.api.upcheck().await {
-                Ok(()) => {
-                    if logging.is_enabled() {
-                        info!(
-                            self.log,
-                            "Execution engine online";
-                        );
-                    }
+    pub async fn upcheck(&self) {
+        let state = match self.api.upcheck().await {
```

Comment:
nothing wrong here specifically but I noticed this [comment in this upcheck() function](https://github.com/sigp/lighthouse/blob/unstable/beacon_node/execution_layer/src/engine_api/http.rs#L597-L602). I want to make sure this hasn't fallen off the radar?

Reply:
Ah yeah, I also had a think about this during the PR. I'm not sure the chain-id checks are necessary, they're a bit of a hangover from "eth1". It was to prevent, say, a mainnet node from unwittingly building it's caches from Goerli.

The "exchange transition config" endpoint should achieve the same thing before the merge, then after the merge it should become obvious very quickly if the nodes are on the wrong chain.

I'm tempted to just leave the TODO there for now, we can pick it up again when we do a `git grep TODO` at some point in the future.